### PR TITLE
Feeder Areas & Parties

### PR DESCRIPTION
**Pull request by grant-m-s**
Feeder areas have been made so that you are only required to define locations for tiles, entities and lights once. As long as the bitmap image is large enough to fit all you areas then it will be repeated a defined number of times.

Party's have also been implemented where a persons party affects which area they spawn in. The player's Party number is set in terminal after ip.

This has been tested with 100 players and 100 ai and the performance seems to hold.

~~Less jank has been commented out for performance, however now the animations are very jank~~

### CODE REVIEW

**Comment by RcColes** [comment]
Going to quickly patch this to fix the underlying componentStage issues, because it's a better solution than removing fix-jank

**Comment by grant-m-s** [comment]
okie dokie t

**Comment by RcColes** [comment]
Right, this could do with a lab test

**Comment by grant-m-s** [comment]
system_less_jank want fully uncommented, also removed playerlocal requirement as all enemies were __jank__ seems to be holding at 60 now, nicely done

File: src/client/src/system/SystemRenderSceneNode.cpp [hunk 1]
```diff
@@ -61,17 +61,17 @@ void SystemRenderSceneNode::setup(irr::scene::ISceneManager *smgr, irr::video::I
...
-			
+
 			sn.spriteDim = m.spriteDim;
 			sn.billboard->getMaterial(0).getTextureMatrix(0).setTextureScale(1.f / sn.spriteDim.y, 1.f / sn.spriteDim.x);
 			sn.updateNeeded = true;
 		}
 	}
 
-	update();
+	// update(playerpos);
```
Comment (AnthonyWharton) [inline src/client/src/system/SystemRenderSceneNode.cpp]: What's this? Should it just be removed if not required

File: src/lib-tempo/include/tempo/network/ID.hpp [hunk 2]
```diff
@@ -72,20 +72,21 @@ enum ComponentID {
...
-	STAGE_ROTATION,     // 15
-	STAGE_TRANSLATION,  // 16
-	TEAM,               // 17
-	TRANSFORM,          // 18
-	WEAPON,             // 19
+	PARTY,              // 9
+	PICKUP,             // 10
+	PLAYER_LOCAL,       // 11
+	PLAYER_REMOTE,      // 12
+	SPIKES,				      // 13
```
Comment (AnthonyWharton) [inline src/lib-tempo/include/tempo/network/ID.hpp]: Spacing ... c'mon

File: src/server/main.cpp [hunk 3]
```diff
@@ -120,72 +120,86 @@ int main(int argc, const char **argv)
...
-	spikes = { { 29,65 } };
-	newButton(world, { { 29,63 },{ 29,67 } }, {}, spikes);
+		for(int j = 0; j < 10; j++){
+		tempo::createMobCreeper(world, glm::ivec2(40 + (i* fheight), 40));
+	}
 
-	spikes = { { 28,65 } };
-	newButton(world, { { 28,63 },{ 28,67 } }, {}, spikes);
+		std::deque<glm::ivec2> path {glm::ivec2(64 + (i* fheight), 68),
+																 glm::ivec2(64 + (i* fheight), 72),
```
Comment (AnthonyWharton) [inline src/server/main.cpp]: The heck is this spacing.

File: src/server/main.cpp [hunk 4]
```diff
@@ -120,72 +120,86 @@ int main(int argc, const char **argv)
...
+																 glm::ivec2(68 + (i* fheight), 72),
+																 glm::ivec2(68 + (i* fheight), 68)};
+		tempo::createMobPatroller(world, path[0], path);
+
+		std::deque<glm::ivec2> path2 {glm::ivec2(11 + (i* fheight), 67),
```
Comment (AnthonyWharton) [inline src/server/main.cpp]: Spacing again.

**Comment by grant-m-s** [comment]
Since commit da02fb5 the framerate has become unusable, we will need to fix or revert

**Pull request merged**